about_binaries.exs

```diff
@@ -31,4 +31,46 @@ defmodule AboutBinaries do
     contatenated = "Hello " <> "world"
     assert contatenated == __?
   end
+
+  def bit_representation(integer) do
+    Integer.to_string(integer, 2) # base 2, AKA "binary"
+  end
+
+  @tag :focus
+  think "Numbers can be represented in base 2 (binary) as a sequence of bits" do
+    assert bit_representation(1) == "1"
+    assert bit_representation(2) == "10"
+    assert bit_representation(3) == "11"
+    assert bit_representation(4) == "100"
+  end
+
+  @tag :focus
+  think "255 is the biggest value we can represent with 8 bits (1 byte)" do
+    assert bit_representation(255) == "11111111"
+    assert String.length(bit_representation(255)) == 8
+    assert bit_representation(256) == "100000000"
+    assert String.length(bit_representation(256)) == 9
+  end
+
+  @tag :focus
+  think "Unicode defines far more than 255 codepoints" do
+    assert ?λ == 955
+    assert ?™ == 8482
+  end
+
+  @tag :focus
+  think "Therefore, most Unicode codepoints must be encoded using more than 1 byte" do
+    assert ?λ == 955
+    assert String.length("λ") == 1
+    assert byte_size("λ") == 2
+    assert "λ" === <<206, 187>>
```

Comment:
>  I might have thought i'd take more than 2 bytes since 955 > 256*2

Each additional bit doubles how many numbers we can represent - with 9 bits, we can have every possible 8-bit value preceded by a 0, and every possible 8-bit value preceded by a 1. So 256 * 2 = 504 possible values in 9 bits, * 2 = 1008 possible values in 10 bits. As easier way to show it is `2 ** 8 = 256` possible values but `2 ** 16 = 65536` possible values. Make sense?

As for the `206, 187`, that's actually sort of a misinterpretation. What we really have is some bits like this:

    utf8_encoding == ["11001110", "10111011"]

When Elixir reads through the binary's bytes and displays the characters, it says "ok, the first one is a UTF8 byte that tells me it's the first of two bytes in a character, because it starts with the header '110'. The rest of that byte is '01110', so I'll hang onto that. The second byte is the second of the two in this character, and sure enough, it starts with the header '10' to confirm that it's a continuation byte. After that, it contains '111011'. So ignoring the headers, I've got '01110' from the first byte and '111011' from the second one. Putting those together gives me '01110111011', which is binary for the number 955 (`String.to_integer("01110111011", 2) => 955`) So these two bytes encode codepoint 955, which [is lambda](http://www.codetable.net/decimal/955). I'll display a lambda!"

That's what it does when it's going to display the characters. But if you just look at the binary, it take each byte separately and treats it as a number. Instead of treating "11001110" as the first byte of a UTF8 codepoint, it just treats it as a binary number, and `String.to_integer("11001110", 2) => 206`. Likewise, the second byte is shown as `String.to_integer("10111011", 2) => 187`.

Whew! 😅